src/app/components/Transactions/TransactionLink.tsx

```diff
@@ -8,22 +8,27 @@ import useMediaQuery from '@mui/material/useMediaQuery'
 import { TrimLinkLabel } from '../TrimLinkLabel'
 import { RouteUtils } from '../../utils/route-utils'
 
-export const TransactionLink: FC<{ alwaysTrim?: boolean; layer: Layer; hash: string }> = ({
-  alwaysTrim,
-  hash,
-  layer,
-}) => {
+export const TransactionLink: FC<{
+  alwaysTrim?: boolean
+  layer: Layer
+  hash: string | undefined
+  eth_hash: string | undefined
+}> = ({ alwaysTrim, hash, eth_hash, layer }) => {
   const theme = useTheme()
   const isMobile = useMediaQuery(theme.breakpoints.down('sm'))
-  const to = RouteUtils.getTransactionRoute(hash, layer)
+  if (!hash && !eth_hash) {
+    throw new Error('Either HASH or ETH_HASH must be provided!')
+  }
+  const to = RouteUtils.getTransactionRoute(eth_hash || hash!, layer)
+  const label = eth_hash ? `0x${eth_hash}` : hash!
```

Comment:
I think even links should contain 0x prefix for evm addresses.

(so maybe we should add 0x prefix in api.ts, instead of everywhere it is used)

Reply:
OK, if we want `0x` in the URL, then it changes everything. I will move the transformation into the API layer....

Reply:
done